src/details/ArborX_Predicates.hpp

```diff
@@ -22,6 +22,11 @@ struct NearestPredicateTag
 struct SpatialPredicateTag
 {};
 } // namespace Details
+namespace Experimental
+{
+struct OrderedIntersectionPredicateTag
+{};
+} // namespace Experimental
```

Comment:
Considering the number of time we do
```C++
  std::is_same<PredicateTag, SpatialPredicateTag>{} ||
    std::is_same<PredicateTag, NearestPredicateTag>{} ||
    std::is_same<PredicateTag, Experimental::OrderedIntersectionPredicateTag>{}
```
we should define a `is_valid_predicate_tag` traits here